bigquery/tests/unit/test_dbapi_connection.py

```diff
@@ -0,0 +1,73 @@
+# Copyright 2016 Google Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+
+import unittest
+
+
+class TestConnection(unittest.TestCase):
+
+    @staticmethod
+    def _get_target_class():
+        from google.cloud.bigquery.dbapi import Connection
+        return Connection
+
+    def _make_one(self, *args, **kw):
+        return self._get_target_class()(*args, **kw)
+
+    def test_ctor(self):
+        from google.cloud.bigquery.dbapi import Connection
+        client = _Client()
+        connection = self._make_one(client=client)
+        self.assertIsInstance(connection, Connection)
+        self.assertIs(connection._client, client)
+
+    def test_connect_wo_client(self):
+        from google.cloud.bigquery import Client
+        from google.cloud.bigquery.dbapi import connect
+        from google.cloud.bigquery.dbapi import Connection
+        connection = connect()
```

Comment:
This test needs to mock out the environment-sniffing such that it runs without any of the "normal" environment variables set.